internal/batches/service.go

```diff
@@ -678,25 +686,47 @@ func (svc *Service) resolveRepositorySearch(ctx context.Context, query string) (
 			}
 		}
 	}
+	searchQuery := setDefaultQueryCount(query)
+	queryWithDefaultCount := searchQuery
+	if svc.features.selectSearchOperator {
+		var err error
+		searchQuery, err = setSelectRepo(searchQuery)
+		if err != nil {
+			return nil, err
+		}
+	}
 	if ok, err := svc.client.NewRequest(repositorySearchQuery, map[string]interface{}{
-		"query":       setDefaultQueryCount(query),
+		"query":       searchQuery,
 		"queryCommit": false,
 		"rev":         "",
 	}).Do(ctx, &result); err != nil || !ok {
 		return nil, err
 	}
 
-	ids := map[string]*graphql.Repository{}
 	var repos []*graphql.Repository
-	for _, r := range result.Search.Results.Results {
-		existing, ok := ids[r.ID]
-		if !ok {
-			repo := r.Repository
-			repos = append(repos, &repo)
-			ids[r.ID] = &repo
-		} else {
-			for file := range r.FileMatches {
-				existing.FileMatches[file] = true
+	if svc.features.selectSearchOperator {
+		repos = make([]*graphql.Repository, 0, len(result.Search.Results.Results))
+		for _, r := range result.Search.Results.Results {
+			if r.Repository.IncludedInSearchQueries == nil {
+				r.Repository.IncludedInSearchQueries = make(map[string]bool)
+			}
+			r.Repository.IncludedInSearchQueries[queryWithDefaultCount] = true
```

Comment:
But if you use `queryWithDefaultCount` here, doesn't that mean that not all file matches will be fetched later on? Before this PR we'd fetch all the FileMatches we could get, but with this change we'd use the default count per repository, right?

Reply:
The "default count" is `9999999` 😬  it's the default for batch changes, not sourcegraph search default

Reply:
Ahhh! `queryWithCount` might be a better name then (I now see though that the name comes from `setDefaultQueryCount`)